zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -398,25 +563,26 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
   final private[zhttp] def execute(a: A): HExit[R, E, B] =
```

Comment:
I think `execute` should be wrapped in a try catch block. And incase of failure, it the function should return an `HExit.die`.

Reply:
Done. Also added a pair of tests to check unexpected exceptions are properly catched and converted to defects.